### ISSUE TITLE
Installation issue: z3 with intel compiler undifined references in lar_solver.cpp

### ISSUE THREAD

**Issue by EthanS94** [LLNL/spack#20522]
### Steps to reproduce the issue

```console
$ spack install z3@4.8.7%intel@19.0.4.243+python arch=linux-rhel7-broadwell
...
util/lp/lp.a(lar_solver.o): In function `lp::lar_solver::solve_with_core_solver()':
lar_solver.cpp:(.text+0x94d0): undefined reference to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)'
util/lp/lp.a(lar_solver.o): In function `lp::lar_solver::prepare_costs_for_r_solver(lp::lar_term const&)':
lar_solver.cpp:(.text+0xc5f9): undefined reference to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)'
util/lp/lp.a(lar_solver.o): In function `lp::lar_solver::move_non_basic_columns_to_bounds()':
lar_solver.cpp:(.text+0x12618): undefined reference to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)'
util/lp/lp.a(lar_solver.o): In function `lp::lar_solver::update_x_and_inf_costs_for_columns_with_changed_bounds_tableau()':
lar_solver.cpp:(.text+0x1423e): undefined reference to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)'
util/lp/lp.a(lar_solver.o): In function `lp::lar_solver::clean_inf_set_of_r_solver_after_pop()':
lar_solver.cpp:(.text+0x2b88e): undefined reference to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)'
util/lp/lp.a(lar_solver.o):lar_solver.cpp:(.text+0x2b953): more undefined references to `lp::lp_primal_core_solver<rational, lp::numeric_pair<rational> >::update_inf_cost_for_column_tableau(unsigned int)' follow
make: *** [z3] Error 1
make: *** Waiting for unfinished jobs....
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.16.0-309-5f2a82118
* **Python:** 3.6.3
* **Platform:** linux-rhel7-broadwell
* **Concretizer:** original
```

### Additional information

I tried with `z3@4.8.9` and did not hit this error. Maybe it's enough to just add version 4.8.9 to z3's package.py file?

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by EthanS94** [comment]
https://github.com/Z3Prover/z3/pull/2899 is what led me to try 4.8.9

**Comment by EthanS94** [comment]
Now that version 4.8.9 is available thanks to #21167 , I am not running into this issue. Closing this
